spec/The-OpenFF-System.md

```diff
@@ -0,0 +1,185 @@
+# The OpenFF System Specification
+
+The OpenFF System provides an internal representation of parametrized systems for molecular simulation.
+
+## Authors and acknowledgments
+
+The OpenFF System specification was designed by the [Open Force Field Initiative](https://openforcefield.org).
+
+Primary contributors include:
+* Matt Thompson `<matt.thompson@openforcefield.org>`
+* Jeff Wagner `<jeffrey.wagner@openforcefield.org>`
+
+## Representation and encodings
+
+The core of the OpenFF System is an internal (i.e., in computer memory) representation of chemical systems. This is in contrast to most models using in computational chemistry, which are based off of an on-disk representation.  Popular serialization formats accessible from Python dictionaries (i.e. XML, JSON and its derivatives) will be supported through a light serialization layer. Serialization to domain-specific formats (`.mol2`, `.pdb`, `.prmtop`, etc.) will be handled with a combination of native writers and conversion to external libraries.
+
+## Primary components
+
+Components will have discrete responsibilities in order to prevent scope creep and ambiguities about what data can be stored where. Wherever possible, components will have lists of "MUST, MAY, MUST NOT" that specify what data is necessary to construct or convert a component, what a component optionally may know, and what it is forbidden from knowing.
+
+TODO: Populated tables like this for each component below
+
+| MUST | MAY |  MUST NOT |
+|---|---|---|
+| Required things | Optional things | Forbidden things |
+
+### Force field data
+
+System parameters (force field parameters applied to a chemical topology) are represented as the sum of individual components (`PotentialHandler`s). Teach term in a potential energy function is expected to be captured by a `PotentialHandler` or combination thereof. These closely mirror the `ParameterHandler`s in OpenFF Toolkit, and may merge in the future.
+
+Each `PotentialHandler` subclass must specify an string-like `expression` that encodes the algebra of its energy evaluation and a collection of `independent_variables` that specify which variables in the expression do not need to be specified by system parameters. The remaining variables are then expected to be specified in a sequence of `Potential` objects stored in the handler.
```

Comment:
Are we sure that everything can be expressed as string-like, rather than  a function that gets evaluated a number of times?  "PME electrostatics short range" is technically a very complicated function, and expressing it as a string could be problematic.   Machine learning potentials are going to be super hard if not impossible to express as strings.

Reply:
Can we have string constants, that define the algebra elsewhere? Is that a good or bad idea?

Reply:
I do think we'll need to have a few different flavors of this class - also for tabulated potentials that do not care about any algebraic form